File: rest_api/src/models/product.rs
use serde_derive::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct Product {
    pub id: Option<i32>,
    pub name: String,
    pub category: String,
    pub price: i32,
    pub description: String,
    pub in_stock: bool,
    pub weight: i32,
    pub discount: Option<i32>,
}   

File: frontend/src/pages/api/product.ts
// src/api/product.ts
import axios from 'axios';

const API_URL = 'http://localhost:8080';

interface Product {
  id: number;
  name: string;
  category: string;
  price: number;
  description: string;
  in_stock: boolean;
  weight: number;
  discount?: number | null;
}

export const getAllProducts = async (): Promise<Product[]> => {
  try {
    const response = await axios.get(`${API_URL}/product`);
    return response.data;
  } catch (error) {
    console.error("Erro ao buscar produtos", error);
    throw error;
  }
};

export const createProduct = async (productData: Product): Promise<Product> => {
  try {
    const response = await axios.post(`${API_URL}/product`, productData);
    return response.data;
  } catch (error) {
    console.error("Erro ao criar produto", error);
    throw error;
  }
};
